pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -407,21 +406,17 @@ private static boolean addReturning(StringBuilder nativeSql, SqlCommandType curr
   }
 
   /**
-   * Converts {@code List<Integer>} to {@code int[]}. Empty and {@code null} lists are converted to
-   * empty array.
+   * Converts {@link IntList} to {@code int[]}. A {@code null} collection is converted to
+   * {@code null} array.
    *
    * @param list input list
    * @return output array
    */
-  private static int[] toIntArray(@Nullable List<Integer> list) {
-    if (list == null || list.isEmpty()) {
-      return NO_BINDS;
-    }
-    int[] res = new int[list.size()];
-    for (int i = 0; i < list.size(); i++) {
-      res[i] = list.get(i); // must not be null
+  private static int @Nullable [] toIntArray(@Nullable IntList list) {
```

Comment:
Is the caller ok with this returning null (rather than an empty array)?

Reply:
CheckerFramework validates that. `NativeQuery` handles `null array` just fine. I added `@Nullable` to the parameter though